WIP: GitHub repository documentation

jaanjah opened the pull request:
Closes: https://github.com/PulsarcGame/Pulsarc/issues/38
This merge request will add the documentation part on GitHub for the project.

Todo list:
- [X] - Info about the game - ABOUT.md
- [x] - Add game logo to README.md
- [x] - Instructions for compiling & running the game
- [x] - Instructions how you can contribute to the game.

For compiling section we need a working copy of the game in master branch.

I think we can postpone the contributing section since there is still no code of conduct in place.

Feedback what to add is appreciated.

jaanjah committed:
Add readme

jaanjah committed:
Fix discord widget

jaanjah committed:
Add ABOUT.md to README.md

dustdustinthewind reviewed (commented):
Good work so far. Thanks! We don't have an "official" logo yet but a lot of us like Makaii's design, feel free to use one of those for now if you like.
![pulsarc_pulse_box](https://user-images.githubusercontent.com/16842122/61655597-bd615580-ac73-11e9-90fc-c758f3a87ec0.png)

This one is my personal favorite x)

jaanjah commented:
The logo itself looks good, but I think a transparent background would look better on the github site.

jaanjah committed:
Add logo

dustdustinthewind committed:
Update ABOUT.md

Slight change to "Sounds cool!" section.
Sorry if I wrote something weird, I'm a little high right now xd.

dustdustinthewind committed:
Update README.md

Fixed image size

dustdustinthewind committed:
Basic Play/Convert Info

As requested from #38

jaanjah commented on README.md:
```diff
@@ -1,2 +1,38 @@
+## About
+[Read more about this project!](docs/ABOUT.md)
+
+## How to Play the Prototype
```
This part should go under ## Compiling in a COMPILING.md file under docs/

jaanjah commented on README.md:
```diff
@@ -1,2 +1,38 @@
+
+**Visual Studio Community** - https://docs.microsoft.com/en-us/visualstudio/releasenotes/vs2017-relnotes
+
+**.NET Core 2.2 SDK (v2.2.108)** - https://dotnet.microsoft.com/download/thank-you/dotnet-sdk-2.2.108-windows-x64-installer
```
Perhaps lets encourage users to use visual studio 2019 since visual studio 2017 won't be getting any more updates.

Adrriii replied:
For now it only complicates the installation, I'm not sure how you can run with 2019. Perhaps you can tell us ?

jaanjah replied:
Visual Studio 2019 works just fine and you can use a newer versions of [.NET Core 2.2 SDK](https://dotnet.microsoft.com/download/dotnet-core/2.2), when we use visual studio 2017, we can only use the 2.2.108 version, in which we are missing updates and security patches from newer versions. Using vs2019 will be good in the long run.

jaanjah replied:
Now that I'm thinking about it, you don't actually need an IDE at all, you can technically just build it from commandline using `dotnet build`

Adrriii replied:
Well, this is a comprehensive guide for all users. Experienced people can have their way ofc. About using newer version of 2.2, as long as it works I guess we can have that.

jaanjah commented on README.md:
```diff
@@ -1,2 +1,38 @@
+**MonoGame 3.7.1 for VisualStudio** - http://community.monogame.net/t/monogame-3-7-1-release/11173
+
+When you have those downloaded and installed, go to the Prototype-branch here: https://github.com/PulsarcGame/Pulsarc/tree/MonoGame-Prototype-Adri. Click the Green "Clone or download" button, then click "Download ZIP" on the bottom right. Extract the .zip into a folder. Open Pulsarc.sln using Visual Studio. When it's done loading click on the little green play icon near the top.
+
```
Do this instead of downloading as zip:
`git clone --recurse-submodules https://github.com/PulsarcGame/Pulsarc` <- has to be merged to master first though.

Adrriii replied:
See my comment on the commit (can be seen on the PR)

jaanjah commented on docs/ABOUT.md:
```diff
@@ -0,0 +1,10 @@
+## Sounds cool! How can I help out?
+[![Discord](https://discordapp.com/api/guilds/486933399425122318/widget.png?style=shield)](https://discord.gg/SYfpvfJ)
+
+Feel free to contribute to the Github, and come join our discord server! Just being a part of the server is support enough! Anyone on the server can give suggestions, and comment on the current progress.
```
Perhaps `Feel free to contribute on GitHub` would make more sense.

jaanjah commented on README.md:
```diff
@@ -1,2 +1,38 @@
+When you have those downloaded and installed, go to the Prototype-branch here: https://github.com/PulsarcGame/Pulsarc/tree/MonoGame-Prototype-Adri. Click the Green "Clone or download" button, then click "Download ZIP" on the bottom right. Extract the .zip into a folder. Open Pulsarc.sln using Visual Studio. When it's done loading click on the little green play icon near the top.
+
+Use the scroll wheel to move the song select up or down. Click on the map you want to play to play it.
+Hit arcs using D - Left, F - Up, J - Down, K - Right. If you want to change the keys, you can edit them starting at line 121 of GameplayEngine.cs, which is found in Pulsarc\UI\Screens\Gameplay.
```
This note is temporary and will be likely changed in later versions of the game, I don't think this should be in README.
Instead of telling users to directly edit code, perhaps we should wait until we have a proper configuration system in place, and have a wiki on how to use it.

dustdustinthewind replied:
Yeah I was iffy writing that. I don't like telling users to code so they can play the prototype either. I'll remove it when I get on my PC.

jaanjah commented on README.md:
```diff
@@ -1,2 +1,38 @@
+
+## How to Add Maps
+Right now the Pulsarc prototype comes with one map to play. However, the prototype can convert osu!mania or Intralism maps into a pulsarc beatmap, that will be added to the Song Select.
+If you've played the Pulsarc Prototype through Visual Studio at least once, you can edit the config.ini found in "Pulsarc\bin\Debug\netcoreapp2.2\config.ini". Open this file in your text editor of choice. Put "Intralism" or "Mania" after "Game= ". Find the path to the folder of the map you want to convert. Copy the path address and then paste it after "Path= ". Save config.ini. In the Pulsarc Son ...
```
Same as I said above ^, This will likely be changed in later versions and I think it's better to wait when we have drag & drop feature in place.

Adrriii replied:
The drag & drop feature would come much later, so this one makes sense to have here. Or maybe i can get started on it soon?

jaanjah commented on README.md:
```diff
@@ -1,2 +1,38 @@
+
+**.NET Core 2.2 SDK (v2.2.108)** - https://dotnet.microsoft.com/download/thank-you/dotnet-sdk-2.2.108-windows-x64-installer
+
+**MonoGame 3.7.1 for VisualStudio** - http://community.monogame.net/t/monogame-3-7-1-release/11173
```
Not needed since we get it through submodules

Adrriii replied:
I'm not sure if this will be added through submodules since these are installed via windows installer

jaanjah replied:
We are using monogame submodule in Wobble, which should have all monogame stuff that we are using.

Adrriii replied:
Yes but compiling and some content related features rely on you having monogame and the monogame pipeline installed on your computer. If you are not on windows it might not be necessary. I'm not sure to what extent you can do stuff without installing, but just to avoid future troubleshooting it'd be best to have all contributors and testers have those installed until we deliver some published builds

dustdustinthewind committed:
Update ABOUT.md

dustdustinthewind committed:
Update README.md

jaanjah commented:
This pull request is stale. Something should be done about this.

What is the missing info thatshould added to this PR and so this could be merged? @FlyingRabidUnicornPig

dustdustinthewind commented:
I got started on a revised readme a week or two ago, but forgot to finish it. Lemme finish that draft right now and we can discuss it.

dustdustinthewind commented:
- Made a new About section (we can probably get rid of About.md)
- Added How to Play and updated How to Add Maps
- Changed "Compiling" to "Build and Run" and included steps to build and run without an IDE.
- Added info to "Contributing"

jaanjah commented on README.md:
```diff
@@ -1,2 +1,50 @@
+## Build and Run
+To build and run Pulsarc, follow these steps:
+- Install the [.NET Core 2.2 SDK](https://dotnet.microsoft.com/download/dotnet-core/2.2)
+- Clone the Pulsarc repository and its submodules ``git clone --recurse-submodules https://github.com/PulsarcGame/Pulsarc.git``
```
```suggestion
- Clone the Pulsarc repository and its submodules `git clone --recurse-submodules https://github.com/PulsarcGame/Pulsarc.git`
```
Double backticks aren't needed :)

jaanjah commented on README.md:
```diff
@@ -1,2 +1,50 @@
+To build and run Pulsarc, follow these steps:
+- Install the [.NET Core 2.2 SDK](https://dotnet.microsoft.com/download/dotnet-core/2.2)
+- Clone the Pulsarc repository and its submodules ``git clone --recurse-submodules https://github.com/PulsarcGame/Pulsarc.git``
+- Build & Run Pulsarc with ``dotnet run --project Pulsarc``
```
```suggestion
- Build & Run Pulsarc with `dotnet run --project Pulsarc`
```

jaanjah commented on README.md:
```diff
@@ -1,2 +1,50 @@
+
+You can make a contribution through a pull request. Don't make requests for the master branch, use the [develop branch](https://github.com/PulsarcGame/Pulsarc). Please have one branch/pull request per feature, and keep pull requests short and sweet. We will let you know if we want to discuss more, or you can bring up a discussion on the Discord server.
+
+When contributing, please follow our (code style)[https://github.com/PulsarcGame/Pulsarc/blob/develop/CODESTYLE.md]. If you have any issues with the code style feel free to discuss on the issues board or our Discord server.
```
I'm pretty sure you can just put `CODESTYLE.md` instead of entire link, since it's located in same repo. would shorten the links quite a bit. Same with `LICENSE` file below

jaanjah commented on README.md:
```diff
@@ -1,2 +1,50 @@
+Pulsarc is currently in active development. Pulsarc is still a young project, and is in an early Alpha state. The core gameplay is here, but many features haven't been realized yet, and there are currently many bugs.
+
+## How to Play
+Head over to our [Releases Page](https://github.com/PulsarcGame/Pulsarc/releases) and download the most recent version. Find and launch Pulsarc.exe to get started. If you want to play the current development build, check out the "Contributing" section below.
```
Add reference link to the `Contributing` section

jaanjah commented:
Commits should probably be squashed into 1 commit, because most of them are just `Update *` at the moment.

dustdustinthewind committed:
Update README.md

used Quaver's readme as a reference: https://github.com/Quaver/Quaver/blob/ui-redesign/README.md

dustdustinthewind committed:
Delete ABOUT.md

all this info is in Readme now

dustdustinthewind reviewed (approved):
I'm going to merge this into master now, thank you for your help Jaan!
